### DISCUSSION TITLE
Can we run large language model to generate text using brainJS?

### DISCUSSION THREAD

**Discussion by linonetwo** [BrainJS/brain.js#873] [Q&A] [answered]
As I ask in https://github.com/BrainJS/brain.js/issues/335#issuecomment-1413416621 

Hi, I'm writing a electronjs based tiddlywiki app https://github.com/tiddly-gittly/TidGi-Desktop that is offline-first, and I want to make it possible to run a large language model (around 1GB is ok I think?).

Can brainJS do text generation now? The size of model is not a problem, because I'm going to use it in an electron app.

**Comment by robertleeplummerjr** [comment, marked as answer]
>Can we run large language model to generate text using brainJS?

Yes. The neural network which does this is the `brain.recurrent.LSTM` class. It however isn't GPU accelerated, so it can be slow if the network is larger than say a hidden layer of size 500+.

>Can brainJS do text generation now?

The same answer as the above.
